File: c/generators/test_rendering_options.c
// SPDX-License-Identifier: MIT
#include "test_utils.h"
#include <SketchUpAPI/model/model.h>
#include <SketchUpAPI/model/rendering_options.h>
#include <SketchUpAPI/model/typed_value.h>
#include <stdio.h>

int main(int argc, char** argv) {
    if (!init_sketchup()) return 1;
    enum SUModelVersion ver = version_from_args(argc, argv);

    SUModelRef model = SU_INVALID;
    SUModelCreate(&model);

    SURenderingOptionsRef options = SU_INVALID;
    SUModelGetRenderingOptions(model, &options);

    // Change a few rendering options
    SUTypedValueRef value = SU_INVALID;
    SUTypedValueCreate(&value);
    
    // Disable edges
    SUTypedValueSetBool(value, false);
    SURenderingOptionsSetValue(options, "EdgeDisplayMode", value);

    // Change background color (as a string or int? Usually keys are strings)
    // "BackgroundColor" is a common key.
    SUTypedValueSetBool(value, true);
    SURenderingOptionsSetValue(options, "DrawGround", value);

    SUTypedValueRelease(&value);

    save_model_with_version(model, "data/rendering_options.skp", ver);

    SUModelRelease(&model);
    term_sketchup();
    return 0;
}

File: python/writer/generators/test_rendering_options.py
# SPDX-License-Identifier: MIT
from pathlib import Path

from .common import run


def generate(skppy, destination: Path) -> None:
    model = skppy.new_model()
    model.rendering_options = skppy.RenderingOptions(
        edge_display_mode=0,
        draw_ground=True,
    )
    skppy.save(model, destination)


if __name__ == "__main__":
    raise SystemExit(run(generate, "rendering_options"))
